tests/test_file_set_processor_jp2_with_ocr.py

```diff
@@ -0,0 +1,83 @@
+from pathlib import Path
+
+import pytest
+
+from dor.providers.file_set_identifier import FileSetIdentifier
+from dor.providers.file_system_file_provider import FilesystemFileProvider
+from dor.providers.operations import (
+    CompressSourceImage,
+    CreateTextAnnotationData,
+    ExtractImageText,
+    ExtractImageTextCoordinates,
+)
+from dor.providers.process_basic_image import (
+    Command,
+    Input,
+    process_basic_image,
+)
+
+
+@pytest.fixture
+def file_set_identifier() -> FileSetIdentifier:
+    return FileSetIdentifier(project_id="collid", file_name="quick-brown.jp2")
+
+
+@pytest.fixture
+def input_path() -> Path:
+    return Path("tests/fixtures/test_basic_copy_jp2_with_ocr")
+
+
+@pytest.fixture
+def output_path() -> Path:
+    file_provider = FilesystemFileProvider()
+    output_path = Path("tests/test_process_image_jp2_with_ocr")
+    file_provider.delete_dir_and_contents(output_path)
+    file_provider.create_directory(output_path)
+    return output_path
+
+
+@pytest.fixture
+def jp2_image_with_ocr_input(input_path: Path) -> Input:
+    commands = [
+        Command(operation=CompressSourceImage, kwargs={}),
+        Command(operation=ExtractImageTextCoordinates, kwargs={}),
+        Command(operation=ExtractImageText, kwargs={}),
+        Command(operation=CreateTextAnnotationData, kwargs={}),
+    ]
+    return Input(file_path=input_path / "quick-brown.jp2", commands=commands)
+
+
+def test_process_creates_source_service_image_for_jp2(
+    file_set_identifier, jp2_image_with_ocr_input, output_path
+):
+    assert process_basic_image(
+        file_set_identifier=file_set_identifier,
+        inputs=[jp2_image_with_ocr_input],
+        output_path=output_path
+    )
+  
+    image_file = output_path / file_set_identifier.identifier /  \
+        "data" / ("quick-brown.function:source.function:service.format:image.jp2")
+    assert image_file.exists()
+
+
+def test_process_creates_ocr_data_files(
+    file_set_identifier, jp2_image_with_ocr_input, output_path
+):
+    assert process_basic_image(
+        file_set_identifier=file_set_identifier,
+        inputs=[jp2_image_with_ocr_input],
+        output_path=output_path
+    )
+
+    text_coordinates_file = output_path / file_set_identifier.identifier /  \
+        "data" / ("quick-brown.function:service.format:text-coordinate.xml")
+    text_coordinates_file.exists()
+
+    plain_text_file = output_path / file_set_identifier.identifier /  \
+        "data" / ("quick-brown.function:service.format:text-plain.txt")
+    plain_text_file.exists()
+
+    annotation_file = output_path / file_set_identifier.identifier /  \
+        "data" / ("quick-brown.function:service.format:text-annotation.json")
+    annotation_file.exists()
```

Comment:
The scenario says JP2 plus service OCR, so I wanted to make sure OCR still runs properly, but didn't know exactly how much to test -- so I just opted for making sure the data files are generated. We have other tests to make sure the OCR operations create technical metadata and event files I guess?

Reply:
I think that's fine.

Does this scenario involve text extraction because I thought you were using the source TIFF with tesseract? Because if text extraction is always starting with the service JPEG2000 _this_ scenario could concern itself just with whether "service" is appended to the file's uses.

Reply:
Right now the OCR operations are using the service JP2 for input. We should discuss whether that's the right approach. But if it is, then I think you're saying this test may not be necessary.